File: Core/Client/src/Common/Interfaces/Conversation.ts
import { Message } from "./Message";
import Profile from "./Profile";
import { User } from "./User";

export class Conversation {
    participants: Array<Profile>;
    messages: Array<Message>;

    constructor(participants: Array<Profile>, messages: Array<Message>) {
        this.participants = participants;
        this.messages = messages;
    }

    public filterParticipants = (user: User): Array<Profile> => this.participants.filter(x => x.userId !== user.userId);
}

File: Core/ChatService/Models/Conversation.cs
using Persistence.Profile.Types.DTO;

namespace ChatService.Models;

public class Conversation
{
    public Conversation(IReadOnlySet<ProfileDTO> participants, IReadOnlySet<ConversationMessage> messages)
    {
        Id = id;
        Participants = participants;
        Messages = messages;
    }

    public IReadOnlySet<ProfileDTO> Participants { get; }
    
    public IReadOnlySet<ConversationMessage> Messages { get; }
}